corehq/apps/app_manager/static/app_manager/js/visit-scheduler.js

```diff
@@ -39,7 +133,35 @@ var VisitScheduler = (function () {
             self.saveButton.fire('change');
         };
 
-        self.formSchedule = FormSchedule.wrap(params.schedule, self);
+        self.validate = function(){
```

Comment:
Is this the first place HQ is doing client-side required fields validation? That's kind of fascinating.